zipkin-autoconfigure/storage-elasticsearch-http/src/main/java/zipkin/autoconfigure/storage/elasticsearch/http/BasicAuthInterceptor.java

```diff
@@ -0,0 +1,59 @@
+
+/**
+ * Copyright 2015-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package zipkin.autoconfigure.storage.elasticsearch.http;
+
+import com.squareup.moshi.JsonReader;
+import okhttp3.Credentials;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+import java.io.IOException;
+
+import static zipkin.moshi.JsonReaders.enterPath;
+
+public class BasicAuthInterceptor implements Interceptor {
+
+    ZipkinElasticsearchHttpStorageProperties es;
+
+    BasicAuthInterceptor(ZipkinElasticsearchHttpStorageProperties es) {
```

Comment:
this isn't dynamic, so best to just initialize the credentials here vs on each request